extension/llm/runner/multimodal_input.h

```diff
@@ -0,0 +1,167 @@
+
+  /**
+   * Check if this input contains text data.
+   * @return true if this input contains text, false otherwise.
+   */
+  bool is_text() const {
+    return std::holds_alternative<std::string>(data_);
+  }
+
+  /**
+   * Check if this input contains image data.
+   * @return true if this input contains an image, false otherwise.
+   */
+  bool is_image() const {
+    return std::holds_alternative<Image>(data_);
+  }
+
+  /**
+   * Get the type of data stored in this input.
+   * @return Type::TEXT if text data, Type::IMAGE if image data.
+   */
+  Type get_type() const {
+    return is_text() ? Type::TEXT : Type::IMAGE;
+  }
+
+  /**
+   * Get the text data from this input.
+   * @return Reference to the stored text string.
+   * @throws std::bad_variant_access if this input doesn't contain text.
+   */
+  const std::string& get_text() const {
+    return std::get<std::string>(data_);
+  }
+
+  /**
+   * Get the text data from this input (mutable version).
+   * @return Mutable reference to the stored text string.
+   * @throws std::bad_variant_access if this input doesn't contain text.
+   */
+  std::string& get_text() {
+    return std::get<std::string>(data_);
+  }
+
+  /**
+   * Get the image data from this input.
+   * @return Reference to the stored Image object.
+   * @throws std::bad_variant_access if this input doesn't contain an image.
+   */
+  const Image& get_image() const {
+    return std::get<Image>(data_);
+  }
+
+  /**
+   * Get the image data from this input (mutable version).
+   * @return Mutable reference to the stored Image object.
+   * @throws std::bad_variant_access if this input doesn't contain an image.
+   */
+  Image& get_image() {
+    return std::get<Image>(data_);
+  }
+
+  /**
+   * Try to get the text data from this input safely.
+   * @return Pointer to the text string if this input contains text, nullptr
+   * otherwise.
+   */
+  const std::string* try_get_text() const {
+    return std::get_if<std::string>(&data_);
+  }
+
+  /**
+   * Try to get the text data from this input safely (mutable version).
+   * @return Pointer to the text string if this input contains text, nullptr
+   * otherwise.
+   */
+  std::string* try_get_text() {
+    return std::get_if<std::string>(&data_);
+  }
+
+  /**
+   * Try to get the image data from this input safely.
+   * @return Pointer to the Image object if this input contains an image,
+   * nullptr otherwise.
+   */
+  const Image* try_get_image() const {
+    return std::get_if<Image>(&data_);
+  }
+
+  /**
+   * Try to get the image data from this input safely (mutable version).
+   * @return Pointer to the Image object if this input contains an image,
+   * nullptr otherwise.
+   */
+  Image* try_get_image() {
+    return std::get_if<Image>(&data_);
+  }
+
+ private:
+  std::variant<std::string, Image> data_;
+};
+
+// Convenience factory functions
+inline MultimodalInput make_text_input(const std::string& text) {
+  return MultimodalInput(text);
```

Comment:
I guess instead of wrapping it with repetitive `MultimodalInput` you can just wrap argument with curly brackets which will call accurate constructor.